jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpConnection.java

```diff
@@ -1556,6 +1557,9 @@ public void succeeded()
             if (_handling.compareAndSet(true, false))
                 return;
 
+            // The handling thread has completed, so we can free up any empty buffer here
+            releaseRequestBufferIfEmpty();
+
```

Comment:
The idea of *not* releasing here is that we can re-use an empty existing buffer for the next request, rather than releasing here, and having to re-acquire it for the next request.